docs/source/tutorial/resolvers.md

```diff
@@ -3,10 +3,370 @@ title: "3. Write your graph's resolvers"
+
+The `cancelTrip` function takes in a `launchId`, and makes a request via the `cancelTrip` method of the user data source to cancel a trip. If a trip is canceled successfully, then the function retrieves the launch that has been canceled and returns an object indicating a success status back to the client.
+
+For both functions, if the request fails, then an object containing a failed success status and message is returned to the client.
+
+The `login` function takes in an email, checks the user table in the database via the `findOrCreateUser` method to verify if the user exists or not. If the user exists or a new user is created, return a `base64` encoding of the user's email.
+
+The `base64` encoding of the user's detail is for obscuring the data. The result is a form of unique string token which we use for authentication.
+
+Now, we need to add a resolver function to take care of the booked status on a launch. Copy the code below and paste it just after the `Mutation` resolver functions.
+
+_src/resolvers.js_
+
+```js
+...
+Launch: {
+  isBooked: async (launch, _, { dataSources }) =>
+    dataSources.userAPI.isBookedOnLaunch({ launchId: launch.id }),
+}
+```
+
+The `isBooked` function makes a request to the `isBookedOnLaunch` method of the `UserAPI` datasource class with the id of a launch. This request confirms whether the launch has been booked by the logged-in user.
+
+Now, let's implement the `User` resolver functions.
+
+Copy the code below and paste it just after the `Launch` resolver function.
+
+_src/resolvers.js_
+
+```js
+...
+User: {
+  trips: async (_, __, { dataSources }) => {
+    // get ids of launches by user
+    const launchIds = await dataSources.userAPI.getLaunchIdsByUser();
+
+    if (!launchIds.length) return [];
+
+    // look up those launches by their ids
+    return (
+      dataSources.launchAPI.getLaunchesByIds({
+        launchIds,
+      }) || []
+    );
+  },
+},
+```
+
+When trips for a particular user are requested for, what data is returned?
+
+The `trips` resolver function works like the steps outlined below:
+
+1. It fetches the ids of all the launches a user has ever booked a trip for via the user data source's `getLaunchIdsByUser()` method. Check the code for the `getLaunchIdsByUser` method in the `UserAPI` class to refresh your memory on how the fetch happens.
+2. If there are no launch ids, an empty array is returned. This means the user has not embarked on any trip.
+3. If there's an array of launch ids, fetch all the launches via their ids.
+
 <h2 id="write-query">Write a query in the playground</h2>
 
+Apollo Server sets up GraphQL Playground so that you can start running queries and exploring schemas quickly. Go ahead and open the graph service URL in a browser, `http://localhost:4000/`.
+
+Start by copying the query below and pasting it in the left side of the playground. After pasting, hit the Play button at the center to get a response.
+
+```js
```

Comment:
Showed a query with an argument. I can't show a mutation at this point, because the two mutations, `bookTrip`, and `cancelTrip` depends on a logged-in user id and the `login` mutation method is added at the end of this section

Reply:
That makes sense! Maybe let's add a `Write a mutation in the playground` section at the bottom that shows them how to set the headers in GraphQL Playground and trigger a mutation?